examples/foundational/41-text-only-webrtc.py

```diff
@@ -0,0 +1,165 @@
+    logger.info(f"Starting bot")
+
+    transport = SmallWebRTCTransport(
+        webrtc_connection=webrtc_connection,
+        params=TransportParams(),
+    )
+
+    # Create an HTTP session for API calls
+    async with aiohttp.ClientSession() as session:
+        llm = OpenAILLMService(api_key=os.getenv("OPENAI_API_KEY"))
+
+        messages = [
+            {
+                "role": "system",
+                "content": "You are a helpful LLM in a WebRTC call. Your goal is to demonstrate your capabilities in a succinct way. Respond to what the user said in a creative and helpful way.",
+            },
+        ]
+
+        context = OpenAILLMContext(messages)
+        context_aggregator = llm.create_context_aggregator(context)
+
+        async def action_llm_append_to_messages_handler(
+            rtvi: RTVIProcessor, service: str, arguments: dict[str, any]
+        ) -> ActionResult:
+            run_immediately = (
+                arguments["run_immediately"] if "run_immediately" in arguments else True
+            )
+
+            if run_immediately:
+                await rtvi.interrupt_bot()
+
+            # We just interrupted the bot so it should be fine to use the
+            # context directly instead of through frame.
+            if "messages" in arguments and arguments["messages"]:
+                mess = arguments["messages"]
+                frame = LLMMessagesAppendFrame(messages=arguments["messages"])
+                await rtvi.push_frame(frame)
+
+            if run_immediately:
+                frame = context_aggregator.user().get_context_frame()
+                await rtvi.push_frame(frame)
+
+            return True
+
+        action_llm_append_to_messages = RTVIAction(
+            service="llm",
+            action="append_to_messages",
+            result="bool",
+            arguments=[
+                RTVIActionArgument(name="messages", type="array"),
+                RTVIActionArgument(name="run_immediately", type="bool"),
+            ],
+            handler=action_llm_append_to_messages_handler,
+        )
+
+        rtvi = RTVIProcessor(config=RTVIConfig(config=[]))
+        rtvi.register_action(action_llm_append_to_messages)
+
+        pipeline = Pipeline(
+            [
+                transport.input(),
+                rtvi,
+                context_aggregator.user(),
+                llm,
+                transport.output(),
+                context_aggregator.assistant(),
+            ]
+        )
+
+        task = PipelineTask(
+            pipeline,
+            params=PipelineParams(
+                allow_interruptions=True,
+                enable_metrics=True,
+            ),
+            observers=[RTVIObserver(rtvi)],
+        )
+
+        @rtvi.event_handler("on_client_ready")
+        async def on_client_ready(rtvi):
+            logger.info("Pipecat client ready.")
+            await rtvi.set_bot_ready()
+            # Update small webrtc UI to just handle text
```

Comment:
what is small webrtc UI? Is this a Pipecat standard? Or are these custom messages we have introduced as part of this example to pair with an example client? Where is that client?

Reply:
thanks. while i did want the answers to that question for myself, my bigger point was that we need documentation here to answer those questions for anyone coming to this example. the existing comment is not enough.